subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskDefinitionIntegrationSpec.groovy

```diff
@@ -17,9 +17,37 @@
 package org.gradle.integtests
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.Requires
 import spock.lang.Unroll
 
+import static org.gradle.util.TestPrecondition.KOTLIN_SCRIPT
+
 class TaskDefinitionIntegrationSpec extends AbstractIntegrationSpec {
+    private static final String CUSTOM_TASK_WITH_CONSTRUCTOR_ARGS = """
+            import javax.inject.Inject
+
+            class CustomTask extends DefaultTask {
+                final String message
+                final int number
+
+                @Inject
+                CustomTask(String message, int number) {
+                    this.message = message
+                    this.number = number
+                }
+
+                @TaskAction
+                void printIt() {
+                    println("\$message \$number")
+                }
+            }
+        """
+
+    private static final String KOTLIN_TASK_CONTAINER_EXTENSION = '''
+            inline
+            fun <reified T : Task> TaskContainer.create(name: String, vararg arguments: Any?) =
```

Comment:
Can you use `vararg arguments: Any` as type instead of `Any?` given the non-null requirement?

Reply:
done